lib/cdo/graphics/certificate_image.rb

```diff
@@ -95,10 +95,15 @@ def create_workshop_certificate_image(image_path, fields)
 # Prepare the given string for using in Image Magick.
 def escape_image_magick_string(string)
   string = string.force_8859_to_utf8
-  # Escape special Image Magick symbols @, %, and \n
-  string = string.gsub(/^@/, '\@')
-  string = string.gsub(/%/, '\%')
-  string = string.gsub(/\\n/, '\\\\\n')
+  # Escape special Image Magick symbols \, @, %, and \n
+  # literal \ replaced with literal \\
+  string = string.gsub(/\\/) {'\\\\'}
```

Comment:
what's the motivation to provide the substitution in a block rather than just
```suggestion
  string = string.gsub(/\\/, '\\\\')
```
?

Reply:
https://code-examples.net/en/q/178846

Reply:
> The problem is that when using sub (and gsub), without a block, ruby interprets special character sequences in the replacement parameter. 

oh my god, it behaves fundamentally differently when in block vs not block mode? :nauseated_face: 

Could you add a comment explaining that, and include the link?

Reply:
:scream: What!?

Reply:
Yeah I will leave a comment. Yeah gsub can have different behavior depending on the type of arguments you give it: https://ruby-doc.org/core-2.6.5/String.html

The way we were using it before (a string argument for the replacement) has special symbols because it can reference the match. So backslashes had to have extra escaping.
```
"hello".gsub(/(?<foo>[aeiou])/, '{\k<foo>}')  #=> "h{e}ll{o}"
```

The block syntax doesn't support the special match references so there is no extra processing of backslashes. For example:
```
"hello".gsub(/./) {|match| match.ord.to_s + ' '}  #=> "104 101 108 108 111 "
```

Here are the `gsub` docs: https://ruby-doc.org/core-2.6.5/String.html#method-i-gsub

Reply:
I added some more explanation to the code. Let me know if you like it. I didn't link directly to that blog post because I worry about it not existing in the future. So hopefully the code comment is enough.